src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -341,6 +350,50 @@ public static byte[] ReadAllBytes(string path)
             }
         }
 
+        private static byte[] ReadAllBytesUnknownLength(FileStream fs)
```

Comment:
Maybe I missed something but it doesn't seem to `throw new IOException(SR.IO_FileTooLong2GB)` in case of `newLength > int.MaxValue` like it is done in `ReadAllBytes` for the case where length is known. And it seems like `DefaultArrayPool.Rent` doesn't throw either

Reply:
I think this stops when Rent throws out-of-memory which will happen when you ask it to allocate more than MaxByteArrayLength (or when there is no more memory).

Reply:
I can't see such limit in `DefaultArrayPool`:
https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs#L108
So it will throw only on some runtime limit or out of memory which seem to be inconsistent with known length case

Reply:
`MaxByteArrayLength` is the max array length supported by .NET.

Reply:
I see. Thanks! That's what I actually didn't realize